lib/msf/ui/console/command_dispatcher/core.rb

```diff
@@ -1650,9 +1666,25 @@ def cmd_sessions(*args)
         end
       end
     when 'killall'
-      print_status("Killing all sessions...")
-      framework.sessions.each_sorted.reverse_each do |s|
-        session = framework.sessions.get(s)
+      if search_term
+        matching_sessions = get_matching_sessions(search_term)
```

Comment:
```suggestion
      if search_term
        matching_sessions = get_matching_sessions(search_term)
      else
        matching_sessions = framework.sessions
      end
```
If you move this logic up here you can remove everything else nested in the `if` statement and it will mean you get the printing out of the table regardless of whether you have a search term or not